aten/src/ATen/functorch/BatchRulesViews.cpp

```diff
@@ -438,6 +438,18 @@ std::tuple<Tensor, optional<int64_t>> view_batching_rule(
   return std::make_tuple(self_.view_symint(size_), 0);
 }
 
+std::tuple<Tensor,optional<int64_t>> view_copy_batch_rule(
+    const Tensor& self,
+    optional<int64_t> self_bdim,
+    c10::SymIntArrayRef size) {
+  auto self_ = moveBatchDimToFront(self, self_bdim);
+  SymDimVector view_size(size);
+  view_size.insert(view_size.begin(), self_.size(0));
```

Comment:
append to front of std::vector is inefficient because it results in the movement of all the other elements in the list by 1. We should prefer to create a SymDimVector, reserve the correct amount of space (size.size() + 1), set the first element, and then set the remaining elements.

Reply:
fixed in last commit